controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -352,15 +359,15 @@ public void getDelegationToken(io.pravega.controller.stream.api.grpc.v1.Controll
                             streamObserver.onCompleted();
                         }
                     });
-        } else {
+        } catch (Exception e) {
             log.error("Controller api failed with authenticator error");
             streamObserver.onError(Status.UNAUTHENTICATED
                     .withDescription("Authentication failed")
                     .asRuntimeException());
         }
     }
 
-    public boolean checkAuthorization(String resource, AuthHandler.Permissions expectedLevel) {
+    private String checkAuthorization(String resource, AuthHandler.Permissions expectedLevel) throws AuthorizationException {
```

Comment:
There isn't any reason for this to return a string. It should be void

Reply:
it is returning an empty string as it is one of the parameters to `authenticateExecuteAndProcessResults`. This is used in cases which does not need a token to be created. It returns an empty string instead of a valid token.